bundle/direct/dresources/adapter.go

```diff
@@ -63,12 +63,9 @@ type IResource interface {
 	// Example: func (r *ResourceVolume) DoCreate(ctx context.Context, newState *catalog.CreateVolumeRequestContent) (string, *catalog.VolumeInfo, error)
 	DoCreate(ctx context.Context, newState any) (id string, remoteState any, e error)
 
-	// DoUpdate updates the resource. ID must not change as a result of this operation. Returns optionally remote state.
-	// If remote state is available as part of the operation, return it; otherwise return nil.
-	// Example: func (r *ResourceSchema) DoUpdate(ctx context.Context, id string, newState *catalog.CreateSchema) (*catalog.SchemaInfo, error)
-	DoUpdate(ctx context.Context, id string, newState any) (remoteState any, e error)
-
 	// [Optional] DoUpdateWithChanges updates the resource with information about changes computed during plan. Returns optionally remote state.
+	// If remote state is available as part of the operation, return it; otherwise return nil.
+	// Example: func (r *ResourceSchema) DoUpdateWithChanges(ctx context.Context, id string, newState *catalog.CreateSchema, changes *deployplan.Changes) (*catalog.SchemaInfo, error)
 	DoUpdateWithChanges(ctx context.Context, id string, newState any, changes *deployplan.Changes) (remoteState any, e error)
```

Comment:
we don't need to to call it DoUpdateWithChanges anymore, we can call it DoUpdate, since that name is free now, right?